protozoo/predictor.py

```diff
@@ -1,23 +1,34 @@
 import torch
+
 from ignite.engine import Events, Engine
+from typing import Optional
 
 from protozoo.tiktorch_config_keys import ModelZooEntry
 
 
-def get_predictor(model_zoo_entry: ModelZooEntry) -> Engine:
-    def predict_batch(predictor: Engine, batch) -> torch.Tensor:
-        print('predict')
-        return predictor.state.model(batch)
+class Predictor(Engine):
+    def __init__(self, model_zoo_entry: ModelZooEntry, model: Optional[torch.nn.Module] = None):
```

Comment:
What's the idea behind having both a model_zoo_entry and an (optional) model as argument?

Reply:
To share models across Engines. For Predictor (and a future Evaluator) this would be needed in particular. E.g. Evaluator gets model from Trainer to do a validation run.